src/sentry/api/endpoints/organization_member_invite/index.py

```diff
@@ -0,0 +1,283 @@
+        )
+        has_teams = bool(result.get("teams"))
+
+        if is_member and members_can_only_invite_to_members_teams and has_teams:
+            requester_teams = set(
+                OrganizationMember.objects.filter(
+                    organization=organization,
+                    user_id=request.user.id,
+                    user_is_active=True,
+                ).values_list("teams__slug", flat=True)
+            )
+            team_slugs = [team.slug for team in result.get("teams", [])]
+            # ensure that the requester is a member of all teams they are trying to assign
+            if not requester_teams.issuperset(team_slugs):
+                return Response(
+                    {"detail": "You cannot assign members to teams you are not a member of."},
+                    status=400,
+                )
+
+        if has_teams and not organization_roles.get(assigned_org_role).is_team_roles_allowed:
+            return Response(
+                {
+                    "email": f"The user with a '{assigned_org_role}' role cannot have team-level permissions."
+                },
+                status=400,
+            )
+
+        with transaction.atomic(router.db_for_write(OrganizationMemberInvite)):
+            # remove any invitation requests for this email before inviting
+            existing_invite = OrganizationMemberInvite.objects.filter(
+                Q(invite_status=InviteStatus.REQUESTED_TO_BE_INVITED.value)
+                | Q(invite_status=InviteStatus.REQUESTED_TO_JOIN.value),
+                email=result["email"],
+                organization=organization,
+            )
+            for omi in existing_invite:
+                omi.delete()
+
+            teams = []
+            for team in result.get("teams", []):
+                teams.append({"id": team.id, "slug": team.slug})
+
+            omi = OrganizationMemberInvite(
+                organization=organization,
+                email=result["email"],
+                role=result["orgRole"],
+                inviter_id=request.user.id,
+                organization_member_team_data=teams,
+            )
+
+            omi.save()
+
+        if settings.SENTRY_ENABLE_INVITES and result.get("sendInvite"):
+            referrer = request.query_params.get("referrer")
+            omi.send_invite_email(referrer)
+            member_invited.send_robust(
+                member=omi, user=request.user, sender=self, referrer=request.data.get("referrer")
+            )
+
+        self.create_audit_entry(
+            request=request,
+            organization_id=organization.id,
+            target_object=omi.id,
+            data=omi.get_audit_log_data(),
+            event=(
+                audit_log.get_event_id("MEMBER_INVITE")
+                if settings.SENTRY_ENABLE_INVITES
+                else audit_log.get_event_id("MEMBER_ADD")
+            ),
```

Comment:
this one should probably just be MEMBER_INVITE, and double check what `get_audit_log_data` on org member returns so we can use that audit log